src/controllers/consumer-v2.ts

```diff
@@ -0,0 +1,368 @@
+  /*
+    #swagger.summary = 'Get a list of all published datasets'
+    #swagger.description = 'This endpoint returns a list of all published datasets.'
+    #swagger.autoQuery = false
+    #swagger.parameters['$ref'] = [
+      '#/components/parameters/language',
+      '#/components/parameters/page_number',
+      '#/components/parameters/page_size'
+    ]
+    #swagger.responses[200] = {
+      description: 'A paginated list of all published datasets',
+      content: {
+        'application/json': {
+          schema: { $ref: "#/components/schemas/DatasetsWithCount" }
+        }
+      }
+    }
+  */
+  logger.info('Listing published datasets...');
+
+  try {
+    const lang = req.language as Locale;
+    const pageNumber = parseInt(req.query.page_number as string, 10) || 1;
+    const pageSize = parseInt(req.query.page_size as string, 10) || DEFAULT_PAGE_SIZE;
+
+    const results = await PublishedDatasetRepository.listPublishedByLanguage(lang, pageNumber, pageSize);
+
+    res.json(results);
+  } catch (err) {
+    logger.error(err, 'Failed to fetch published dataset list');
+    next(new UnknownException());
+  }
+};
+
+export const getPublishedDatasetById = async (req: Request, res: Response): Promise<void> => {
+  /*
+    #swagger.summary = "Get a published dataset's metadata"
+    #swagger.description = 'This endpoint returns all consumer metadata for a published dataset.'
+    #swagger.parameters['$ref'] = [
+      '#/components/parameters/language',
+      '#/components/parameters/dataset_id'
+    ]
+    #swagger.responses[200] = {
+      description: 'A json object containing all metadata for a published dataset',
+      schema: { $ref: "#/components/schemas/Dataset" }
+    }
+  */
+  const language = req.query.language ? (req.query.language as Locale) : ('en-GB' as Locale);
+  const datasetDTO = await DatasetDTO.fromDatasetId(res.locals.datasetId, language);
+  res.json(datasetDTO);
+};
+
+export const getPublishedRevisionById = async (req: Request, res: Response): Promise<void> => {
+  const language = req.query.language ? (req.query.language as Locale) : ('en-GB' as Locale);
+  const revisionDto = FullRevision.fromRevision(res.locals.revision, language);
+  res.json(revisionDto);
+};
+
+async function apiSetup(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<{
+  dataset: Dataset;
+  publishedRevision?: Revision;
+  sort?: string[];
+  format?: ConsumerOutFormats;
+  pageNumber?: number;
+  pageSize?: number;
+  language: Locale;
+  errors: boolean;
+}> {
+  const dataset = await PublishedDatasetRepository.getById(res.locals.datasetId, withPublishedRevision);
+  const publishedRevision = dataset.publishedRevision;
+  const language = req.language as Locale;
+
+  if (!publishedRevision) {
+    next(new NotFoundException('errors.no_revision'));
+    return {
+      dataset,
+      language,
+      errors: true
+    };
```

Comment:
This is an unusual pattern for an express app - the `apiSetup` function looks like a request middleware signature `(req, res, next)` but then it doesn't function like one because you're calling it inside an existing request handler like a normal function, and then returning an object as well as calling `next()` with an exception... 

I'm not sure of the actual flow here - what happens to the NotFoundException?

It would be better just to throw the exception and then catch it in the calling function.